core/src/main/java/org/ethereum/beacon/core/types/Gwei.java

```diff
@@ -33,6 +33,23 @@ public Gwei minus(Gwei subtrahend) {
     return new Gwei(super.minus(subtrahend));
   }
 
+  public Gwei minusSat(Gwei subtrahend) {
+    if (this.compareTo(subtrahend) < 0) {
+      return Gwei.ZERO;
+    } else {
+      return minus(subtrahend);
+    }
+  }
+
+  public Gwei plusSat(Gwei addend) {
+    Gwei res = this.plus(addend);
+    if (res.lessEqual(res) && addend.greater(Gwei.ZERO)) {
```

Comment:
condition is always true and looks erroneous

Reply:
another good catch!